src/inference/src/ie_core.cpp

```diff
@@ -732,9 +736,14 @@ class CoreImpl : public ie::ICore, public std::enable_shared_from_this<ie::ICore
 
         std::lock_guard<std::mutex> lock(pluginsMutex);
         auto deviceName = pluginName;
+        if (deviceName == ov::DEFAULT_DEVICE_NAME)
+            deviceName = "AUTO";
```

Comment:
What if we have a single device in pluginRegistry? Seems that this device should be defaut.

Reply:
Yes, This PR just specifies a default device when users didn't select any device.

Reply:
@apankratovantonp The default device name in this PR is used for situation which have multiple devices include AUTO.  
Let's use other PRs for other situation like single device you mentioned or multiple devices without AUTO.

Reply:
What if we don't have auto plug-in installed on the system? Can we provide more detailed message in the error in this case?

Reply:
Detailed message in this case has been provided as below line showing. 
https://github.com/openvinotoolkit/openvino/blob/63121e28ca6539f0d372e34ce6f7596ac4a71a6d/src/inference/src/ie_core.cpp#L744

Reply:
ok, but what if AUTO device is in registry (OpenVINO is compiled with ENABLE_AUTO), but I have not installed AUTO plugin during installation.

Reply:
@ilya-lavrenov  currently it will throw load library runtime exception as below, this is not captured and handled clearly, we can improve in a seperate PR:
Cannot load library 'libov_auto_plugin.so: libov_auto_plugin.so: cannot open shared object file: No such file or directory
@wangleis @yangwang201911  @peterchen-intel, we can consider catch the runtime error in the loadnetwork, and report detailed clear error message to customer

Reply:
as pointed just right by @ilya-lavrenov the situations when AUTO is **NOT** available (e.g. user sets the ENABLE_AUTO to OFF in the cmake), yet other plugins (like CPU) are available. This case not an error and we should not through.

Specifically the logic should check if AUTO is available, then AUTO is default, if not (but the CPU is availabale, then the CPU is default), finally if there is only one available device (e.g. case of MYX plugged to the RaspPi) this is the default device...

Reply:
then we need to define an order for the default devices? like AUTO>CPU>GPU etc.

Reply:
I suppose that if NO AUTO device, we should not guess CPU, etc.
We just need to provide a clear error message what happens

Reply:
@ilya-lavrenov Current exception message in L744 claimed below situation:
> 1. Application do not set device. "No device is provided"
> 2. IE use AUTO as default device. "so AUTO device is used by default"
> 3. AUTO device is not loaded. "which failed loading"

We do not mention the root causes in this situation because there may be several root causes:
> 1. User do not install AUTO during installation
> 2. User installed AUTO. libov_auto_plugin.so do not exist.
> 3. User installed AUTO. libov_auto_plugin.so cannot be loaded.

May I know if you expect more clear message for each root cause?

For other situation, let's discuss in separated PR.

Reply:
@myshevts @ilya-lavrenov Using only one available device without AUTO as default device is new feature in IE Core. Let's confirm if it is available requirement first. If yes, we can close in another PR.

Reply:
> @myshevts @ilya-lavrenov Using only one available device without AUTO as default device 
This is reasonable requirement, that if AUTO is not available (e.g. not enabled in the cmake) the impl falls back to smth else (e.g. CPU, or otherwise single device).
Yet, since this is again some way of logic, shall we rather add the AUTO to the core (so it is always available) so we use single code-path for the device selection? @ilya-lavrenov, what do you think?

Reply:
> Yet, since this is again some way of logic, shall we rather add the AUTO to the core (so it is always available)

works for me, but need to think how to do it.